### PR TITLE
Implemented small change to markdown formatting, and added docs

### PR DESCRIPTION
**Pull request by auddithio**
# *Markdown formatting + documentation*

## :recycle: Current situation & Problem
The current SpeziViews module were not all well documented equally, as per the documentation requirements set forth in the [Stanford Spezi Documentation Guide.](https://swiftpackageindex.com/stanfordspezi/spezi/1.2.1/documentation/spezi/documentation-guide#Landing-Page). Also, the markdown formatting was restricted to `inlineOnly` changes, which would not render formatting for larger changes like titles, which was the issue our app's consent doc ran into.

## :gear: Release Notes 
- Changed markdown formatting to `.full` to incorporate the full breadth of document markdowns
- Added documentation for the Localizable strings section in text

## :books: Documentation
- Added documentation is line with the [Spezi Documentation Guide](https://swiftpackageindex.com/stanfordspezi/spezi/1.2.1/documentation/spezi/documentation-guide#Landing-Page)

## :white_check_mark: Testing
- For the markdown formatting, used the `AttributedString` framework to render markdown documents locally, and it seemed to render properly. Not sure what edge cases the Spezi team wanted to account for by setting it to `inline`, so further testing for the full template application is needed.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziViews/Views/Text/MarkdownView.swift [hunk 1]
```diff
@@ -95,14 +96,16 @@ public struct MarkdownView: View {
     /// Parses the incoming markdown and handles the view's error state management.
     /// - Parameters:
     ///   - markdown: A `Data` instance containing the markdown file in an utf8 representation.
-    ///
+    ///   Example: Data(
+    ///   " # This is a markdown example
+    ///    *This should be italiced* and **this bolded**").utf8
```
Comment (Supereg) [inline Sources/SpeziViews/Views/Text/MarkdownView.swift:101]: If this is a code example, make sure to wrap them into markdown `swift` code block, as you did for the `TextContent´ type as well.
This makes sure that inline code appears with proper syntax highlighting. Make sure to properly format the code according to your swiftlint code style rules.

**Review by Supereg — CHANGES_REQUESTED** [review]
Thank you for improving the documentation of SpeziViews. Thanks for the PR.

I only had small comment regarding a code example on one of the documentation blocks. Otherwise, the PR looks good 👍

**Comment by auddithio** [comment]
made some changes based on feedback and after talking to Spezi staff!

File: Sources/SpeziViews/Views/Text/MarkdownView.swift [hunk 2]
```diff
@@ -95,14 +96,18 @@ public struct MarkdownView: View {
     /// Parses the incoming markdown and handles the view's error state management.
     /// - Parameters:
     ///   - markdown: A `Data` instance containing the markdown file in an utf8 representation.
-    ///
+    ///   ```swift
+    ///   // Example
+    ///   Data(" # This is a markdown example
+    ///    *This should be italiced* and **this bolded**").utf8
+    ///    ```
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Views/Text/MarkdownView.swift:103]: This code example does not compile, if you want to create a String with line breaks you will have to use a multi-line string interval: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/stringsandcharacters/#Multiline-String-Literals

In addition, the code example does not call the API.

Comments like `//Example` should be removed and I would suggest to rather add an explanation around the example block.

Last but not least, I would move the example above the parameters section of the code documentation.

File: Sources/SpeziViews/Views/Text/MarkdownView.swift [hunk 3]
```diff
@@ -95,14 +96,18 @@ public struct MarkdownView: View {
...
     /// - Returns: Parsed Markdown as an `AttributedString`
     @MainActor private func parse(markdown: Data) -> AttributedString {
         state = .processing
         
         guard let markdownString = try? AttributedString(
                 markdown: markdown,
-                options: .init(interpretedSyntax: .inlineOnlyPreservingWhitespace)
+                options: .init(interpretedSyntax: .inlineOnly)
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Views/Text/MarkdownView.swift:110]: What is the reason for this change? Wouldn't it make sense to switch to `.full`?

File: Sources/SpeziViews/Views/Text/TextContent.swift [hunk 4]
```diff
@@ -8,11 +8,24 @@
...
+/// The `TextContent` enum represents either a plain string or a localized string resource.
+///
+/// Use this enum to encapsulate text content in your application, allowing flexibility in handling localized and non-localized strings.
+///
+/// ```swift
+/// // Example usage:
+/// let content: TextContent = .localized(LocalizedStringResource("HELLO_SPEZI", bundle: .main))
+/// let localizedString = content.localizedString(for: .current)
+/// print(localizedString) // Prints the localized string for the current locale.
+/// ```
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Views/Text/TextContent.swift:20]: Thank you for the added documentation!

The same comments about the example also apply here as well.

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
@auddithio Thank you for the PR; I had a few further comments and suggestions that would need to be addressed before we merge the PR 👍

**Pull request closed without merging**